jdaviz/configs/cubeviz/plugins/mixins.py

```diff
@@ -129,9 +129,11 @@ def slice_values(self):
                 except (AttributeError, KeyError):
                     continue
 
-                data = np.asarray(
-                    data_comp.data.take(0, take_inds[0]).take(0, take_inds[1]),
-                    dtype=float)
+                data = data_comp[tuple(data_slice)]
+
+                if not isinstance(data, np.ndarray) or data.dtype.kind not in 'if':
```

Comment:
Because of this the data array will sometimes be integer, which I think we should ensure works - it is wasteful to convert it to float, and in most cases integer images/cubes should work just fine.

Reply:
Are you saying that ideal the other places in this PR where you added `float(thing)` ideally shouldn't be needed, or those are the things you put in to address this comment about integer arrays working? Either way I think this is ok to get in, just trying to understand if you're advocating for further changes.